chia/wallet/wallet_node.py

```diff
@@ -643,10 +638,23 @@ async def receive_and_validate(inner_state: CoinState, inner_idx: int):
                 return False
             while num_concurrent_tasks >= target_concurrent_tasks:
                 await asyncio.sleep(1)
+                if self._shut_down:
```

Comment:
this is great, i was thinking we should do this after efishent reported his issue

Reply:
Extracted to https://github.com/Chia-Network/chia-blockchain/pull/10384 so it can move forward.